### PR TITLE
Add `GridColumnsButton`

### PR DESCRIPTION
**Pull request by andrearutrecht**
`GridColumnsButton` based on MUI `GridToolbarColumnsButton` is added. Stying is according to current Figma design. Start Icon can be overwritten.

---

## PR Checklist

-   [x] Verify if the change requires a changeset. See [CONTRIBUTING.md](https://github.com/vivid-planet/comet/blob/HEAD/CONTRIBUTING.md)
-   [x] Link to the respective task if one exists: COM-654
-   [x] Provide screenshots/screencasts if the change contains visual changes

<details><summary>Video:</summary>
<p>

https://github.com/vivid-planet/comet/assets/122883866/bb3fef01-6c01-4f97-b280-ff8db035336d

</p>
</details>

### CODE REVIEW

**Review by jamesricky — APPROVED** [review]

**Comment by nsams** [comment]
I think this button should be a menu item behind the dropdown menu as it won't be commonly used.

File: packages/admin/admin/src/dataGrid/GridColumnsButton.tsx [hunk 1]
```diff
@@ -0,0 +1,23 @@
...
+type Props = Omit<React.ComponentProps<typeof GridToolbarColumnsButton>, "onResize" | "onResizeCapture">;
+
+export function GridColumnsButton(props: Props) {
+    return (
+        <GridToolbarColumnsButton
+            startIcon={<Columns4 />}
+            variant="outlined"
+            color="info"
+            onResize={undefined}
+            onResizeCapture={undefined}
```
Comment (johnnyomair) [inline packages/admin/admin/src/dataGrid/GridColumnsButton.tsx:17]: Why do you need to override this?

**Review by johnnyomair — COMMENTED** [review]
Do you plan to add this to the generator as well? Should this be enabled by default?

**Comment by jamesricky** [comment]
> I think this button should be a menu item behind the dropdown menu as it won't be commonly used.

The UX Team seems to disagree, they specifically added this button to the design 🙂

**Comment by johnnyomair** [comment]
@andrearutrecht what's the status here?

**Comment by andrearutrecht** [comment]
As far as I know, the UX discussion has not been resolved yet. At the time of discussion, the context menu button did not exist outside of DAM. Should the columns button stay like it is now, or should I wait until this component exists and move it in there? @jamesricky @nsams

**Comment by andrearutrecht** [comment]
In discussion with @jamesricky we decided the button should reflect the design and not move inside a menu.

**Comment by jamesricky** [comment]
> As far as I know, the UX discussion has not been resolved yet. At the time of discussion, the context menu button did not exist outside of DAM. Should the columns button stay like it is now, or should I wait until this component exists and move it in there? @jamesricky @nsams

Since the button explicitly exists in the design, we should add it 🙂

**Comment by andrearutrecht** [comment]
That means the PR stays as is. Is that ok with you, @johnnyomair ?

**Review by jamesricky — APPROVED** [review]

File: packages/admin/admin/src/dataGrid/GridColumnsButton.tsx [hunk 2]
```diff
@@ -0,0 +1,23 @@
...
+import * as React from "react";
+import { FormattedMessage } from "react-intl";
+
+import { messages } from "../messages";
+
+type Props = Omit<React.ComponentProps<typeof GridToolbarColumnsButton>, "onResize" | "onResizeCapture">;
+
+export function GridColumnsButton(props: Props) {
+    return (
+        <GridToolbarColumnsButton
```
[the remark this answers is not in the document]
Comment (andrearutrecht) [reply packages/admin/admin/src/dataGrid/GridColumnsButton.tsx:12]: `GridFilterButton` uses MUIs `useGridApiContext` to open the filter panel. However, there is no function for the column panel in the Context, so I had to use MUIs button directly.

**Review by johnnyomair — APPROVED** [review]

**Pull request merged**